game/graphics/opengl_renderer/OpenGLRenderer.cpp

```diff
@@ -749,16 +751,16 @@ void OpenGLRenderer::do_pcrtc_effects(float alp,
 
     glBindFramebuffer(GL_READ_FRAMEBUFFER, window_blit_src->fbo_id);
     glBindFramebuffer(GL_DRAW_FRAMEBUFFER, 0);
-    glBlitFramebuffer(0,                        // srcX0
-                      0,                        // srcY0
-                      window_blit_src->width,   // srcX1
-                      window_blit_src->height,  // srcY1
-                      0,                        // dstX0
-                      0,                        // dstY0
-                      window_fbo.width,         // dstX1
-                      window_fbo.height,        // dstY1
-                      GL_COLOR_BUFFER_BIT,      // mask
-                      GL_LINEAR                 // filter
+    glBlitFramebuffer(0,                                                          // srcX0
+                      0,                                                          // srcY0
+                      window_blit_src->width,                                     // srcX1
+                      window_blit_src->height,                                    // srcY1
+                      render_state->draw_offset_x,                                // dstX0
+                      render_state->draw_offset_y,                                // dstY0
+                      render_state->draw_offset_x + render_state->draw_region_w,  // dstX1
+                      render_state->draw_offset_y + render_state->draw_region_h,  // dstY1
+                      GL_COLOR_BUFFER_BIT,                                        // mask
+                      GL_LINEAR                                                   // filter
```

Comment:
I think this will not work if you go into windowed mode, resize the window to be not-16:9, then select aspect ratio 16:9. The reason for the "squash" thing earlier is so this blit has the same size source/destination, which ends up being way faster, and is required to resolve the multi-sampled stuff.

Reply:
It does work fine:

![image](https://user-images.githubusercontent.com/7569514/179641932-84141582-5279-4aee-b9b9-9adc2430267c.png)

Not sure about performance though.

Reply:
Hm, if I do:
- delete pc-settings.txt
- start game, it goes to fullscreen
- set to windowed
- resize window
- aspect ratio 16x9
- resize window
I get a black screen with this error:
```
[2022-07-18 21:14:14:072] [error] [] OpenGL error 0x502 S8246 T824C: GL_INVALID_OPERATION error generated. Source and destination dimensions must be identical with the current filtering modes.
```

which is from the `glGlitFramebuffer` with `0, 0, 864, 778, 0, 146, 864, 632` as the arguments.

Reply:
Ah, it does happen if you have MSAA enabled.

Reply:
![image](https://user-images.githubusercontent.com/7569514/179643718-682d6bba-fafd-44f0-8c0c-af1cacc7a123.png)

There we go. 👍